.github/actions/setup/setup.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -le
+
+shopt -s extglob
+
+die () {
+  [[ -n $1 ]] && echo "$*" >&2
+  exit 1
+}
+
+print_output() {
+  [[ -n $1 ]] || die "${FUNCNAME[0]}: missing argument"
+  echo "name=$1::${@:2}"
+  echo "::set-output name=$1::${@:2}"
+}
+
+for i in INPUT_MINIMAL INPUT_OWN_GMX INPUT_REGRESSION_TESTING; do
+  [[ "${!i}" = @(true|false) ]] || die "value of $i is ${!i}, excepted 'true' or 'false'"
+  echo "$i='${!i}'"
+done
+[[ -n ${INPUT_DISTRO} ]] || die "value of INPUT_DISTRO was empty"
+for i in INPUT_DISTRO INPUT_CMAKE_BUILD_TYPE INPUT_TOOLCHAIN INPUT_COVERAGE INPUT_CTEST_ARGS INPUT_CMAKE_ARGS; do
+  echo "$i='${!i}'"
+done
+
+cmake_args=( -DCMAKE_VERBOSE_MAKEFILE=ON -DENABLE_TESTING=ON )
+if [[ ${INPUT_CMAKE_BUILD_TYPE} ]]; then
+  cmake_args+=( -DCMAKE_BUILD_TYPE=${INPUT_CMAKE_BUILD_TYPE} )
+fi
+if [[ ${INPUT_TOOLCHAIN} = "gnu" ]]; then
+  cmake_args+=( -DCMAKE_CXX_COMPILER=g++ -DCMAKE_C_COMPILER=gcc )
+elif [[ ${INPUT_TOOLCHAIN} = "clang" ]]; then
+  cmake_args+=( -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang -GNinja )
+else
+  die "Unknown INPUT_TOOLCHAIN"
+fi
+if [[ ${INPUT_COVERAGE} && ${INPUT_COVERAGE} != "false" ]]; then
+  cmake_args+=( -DENABLE_COVERAGE_BUILD=ON )
+  cov_tag=true
+else
+  cov_tag=false
+fi
+
+if ${INPUT_MODULE}; then
+  cmake_args+=( -DMODULE_BUILD=ON -DCMAKE_INSTALL_PREFIX=$HOME/votca.install )
+else
+  cmake_args+=( -DCMAKE_INSTALL_PREFIX=/usr )
+fi	
+if ${INPUT_OWN_GMX}; then
+  cmake_args+=( -DBUILD_OWN_GROMACS=ON -DENABLE_WARNING_FLAGS=OFF -DENABLE_WERROR=OFF )
+else
+  cmake_args+=( -DENABLE_WERROR=ON )
+fi
+if ${INPUT_MINIMAL}; then
+  cmake_args+=( -DCMAKE_DISABLE_FIND_PACKAGE_HDF5=ON -DCMAKE_DISABLE_FIND_PACKAGE_FFTW3=ON -DCMAKE_DISABLE_FIND_PACKAGE_MKL=ON -DBUILD_MANPAGES=OFF -DCMAKE_DISABLE_FIND_PACKAGE_GROMACS=ON -DBUILD_XTP=OFF )
+else
+  cmake_args+=( -DBUILD_CSGAPPS=ON -DBUILD_XTP=ON -DBUILD_CSG_MANUAL=ON )
+fi
+
+if ${INPUT_MINIMAL} || [[ ${INPUT_DISTRO} = ubuntu@(|_rolling|_devel) ]];  then
+  cmake_args+=( -DENABLE_REGRESSION_TESTING=OFF )
+else
+  cmake_args+=( -DENABLE_REGRESSION_TESTING=${INPUT_REGRESSION_TESTING} )
+fi
+cmake_args+=( ${INPUT_CMAKE_ARGS} )
+print_output "cmake_args" "${cmake_args[@]}"
+
+cache_key="ccache-${INPUT_DISTRO}-${INPUT_TOOLCHAIN}-${INPUT_CMAKE_BUILD_TYPE}-minimal-${INPUT_MINIMAL}-owngmx-${owngmx}-module-${INPUT_MODULE}-coverage-${cov_tag}"
+print_output "cache_restore_key" "${cache_key}"
+print_output "cache_key" "${cache_key}-$(date +%s)"
+
+if [[ ${INPUT_DISTRO} = ubuntu* || ${INPUT_DISTRO} = opensuse ]] || ${INPUT_MODULE}; then
```

Comment:
so sphinx is build via fedora?

Reply:
I haven't tested this opensuse yet, excluding ubuntu is basically #318. I will enable this on more distro down that read.

Will add a comment.

Reply:
Comment added and testing Opensuse currently.

Reply:
Building Spinx on OpenSuse actually works now.